presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnChecksum.java

```diff
@@ -63,19 +101,44 @@ public boolean equals(Object obj)
         }
         ArrayColumnChecksum o = (ArrayColumnChecksum) obj;
         return Objects.equals(checksum, o.checksum) &&
+                Objects.equals(floatingPointChecksum, o.floatingPointChecksum) &&
                 Objects.equals(cardinalityChecksum, o.cardinalityChecksum) &&
                 Objects.equals(cardinalitySum, o.cardinalitySum);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(checksum, cardinalityChecksum, cardinalitySum);
+        return Objects.hash(checksum, floatingPointChecksum, cardinalityChecksum, cardinalitySum);
     }
 
     @Override
     public String toString()
     {
-        return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        if (Objects.isNull(floatingPointChecksum.getSum())) {
+            return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        }
+        else {
+            return format(
+                    "%s, cardinality_checksum: %s, cardinality_sum: %s",
+                    floatingPointChecksum.toString(),
+                    cardinalityChecksum,
+                    cardinalitySum);
+        }
+    }
+
+    public static FloatingPointColumnChecksum.ValidateResult validateFloatingPoint(
```

Comment:
it's weird to me that this logic is over here, but the rest of the validation logic is in ArrayColumnValidator.  I would do one or the other.

Reply:
Moved validate() logic to Validate classes.